target_blob_storage/__init__.py

```diff
@@ -44,42 +44,48 @@ def parse_args():
     overwrite = config.get("overwrite", False)
-    
-    if connect_string:
-        blob_service_client = BlobServiceClient.from_connection_string(connect_string)
-    else:
-        account_name = config.get('account_name')
-        account_key = config.get('account_key')
-
-        sas_token = generate_account_sas(
-            account_name = account_name,
-            account_key = account_key,
-            resource_types=ResourceTypes(object=True),
-            permission=AccountSasPermissions(read=True,add=True,create=True,write=True,delete=True,list=True),
-            expiry=datetime.utcnow() + timedelta(hours=1))
-
-        blob_service_client = BlobServiceClient(account_url=f"https://{account_name}.blob.core.windows.net", credential=sas_token)
-
-    for root, dirs, files in os.walk(local_path):
-        for file in files:
-            file_path = os.path.join(root, file)
-            remote_file_path = file_path.replace(local_path, target_path)
-            blob_client = blob_service_client.get_blob_client(container=container_name, blob=remote_file_path)
-
-            # Upload the created file
-            with open(file_path, "rb") as data:
-                logger.info(f"Uploading: {container_name}:{remote_file_path}")
-                blob_client.upload_blob(data, overwrite=overwrite)
-
-    logger.info(f"Data exported.")
+
+    try:
+        if connect_string:
+            blob_service_client = BlobServiceClient.from_connection_string(connect_string)
+        else:
+            account_name = config.get('account_name')
+            account_key = config.get('account_key')
+
+            sas_token = generate_account_sas(
+                account_name=account_name,
+                account_key=account_key,
+                resource_types=ResourceTypes(object=True),
+                permission=AccountSasPermissions(read=True, add=True, create=True, write=True, delete=True, list=True),
+                expiry=datetime.utcnow() + timedelta(hours=1))
+
+            blob_service_client = BlobServiceClient(account_url=f"https://{account_name}.blob.core.windows.net", credential=sas_token)
+
+        for root, dirs, files in os.walk(local_path):
+            for file in files:
+                file_path = os.path.join(root, file)
+                remote_file_path = file_path.replace(local_path, target_path)
+                blob_client = blob_service_client.get_blob_client(container=container_name, blob=remote_file_path)
+
+                try:
+                    with open(file_path, "rb") as data:
+                        logger.info(f"Uploading: {container_name}:{remote_file_path}")
+                        blob_client.upload_blob(data, overwrite=overwrite)
+                except Exception as e:
+                    logger.error(f"Failed to upload {file_path} to {remote_file_path}: {e}")
+
+    except Exception as e:
+        logger.error(f"An error occurred during the export process: {e}")
```

Comment:
This would just consume the exception and not raise it. It should be `logger.exception`
with a `raise` after that